test/StringUtilsTest.cpp

```diff
@@ -130,3 +131,32 @@ TEST(StringUtilsTest, listToString) {
                         goThroughVectorGenerator(multiValueVector),
                         goThroughVectorGenerator(multiValueVector), ",");
 }
+
+TEST(StringUtilsTest, addIndentation) {
+  // The input strings for testing.
+  static constexpr std::string_view withoutLineBreaks = "Hello\tworld!";
+  static constexpr std::string_view withLineBreaks = "\nHello\nworld\n!";
+
+  // No indentation wanted, should cause an error.
+  ASSERT_ANY_THROW(ad_utility::addIndentation(withoutLineBreaks, 0));
+
+  // Testing identation level 1, 5 and 10.
+  ASSERT_EQ("    Hello\tworld!",
+            ad_utility::addIndentation(withoutLineBreaks, 1));
+  ASSERT_EQ("                    Hello\tworld!",
+            ad_utility::addIndentation(withoutLineBreaks, 5));
+  ASSERT_EQ("                                        Hello\tworld!",
+            ad_utility::addIndentation(withoutLineBreaks, 10));
+
+  ASSERT_EQ("    \n    Hello\n    world\n    !",
+            ad_utility::addIndentation(withLineBreaks, 1));
+  ASSERT_EQ(
+      "                    \n                    Hello\n                    "
+      "world\n                    !",
+      ad_utility::addIndentation(withLineBreaks, 5));
+  ASSERT_EQ(
+      "                                        \n                              "
+      "          Hello\n                                        world\n        "
+      "                                !",
+      ad_utility::addIndentation(withLineBreaks, 10));
+}
```

Comment:
The tests look fine for the given behavior, but maybe we should consider the more generic interface suggested above.